subprojects/jvm-services/src/main/java/org/gradle/jvm/toolchain/internal/JavaInstallationRegistry.java

```diff
@@ -119,6 +125,16 @@ private InstallationLocation canonicalize(InstallationLocation location) {
         }
     }
 
+    private InstallationLocation maybeGetEnclosedInstallation(InstallationLocation location) {
```

Comment:
Yeah I was also considering the change in ordering and behavior here, but I think the new way is more correct. Previously we probed the enclosed jre but later reported the enclosing jdk's path when querying the location via the toolchain. This behavior seems wrong. This could potentially lead to discrepancies. The new way ensures we are probing and reporting information on the actual java home path we plan to use.

Reply:
Added a test to verify this behavior